googleapiclient/model.py

```diff
@@ -140,11 +142,11 @@ def request(self, headers, path_params, query_params, body_value):
     query = self._build_query(query_params)
     headers['accept'] = self.accept
     headers['accept-encoding'] = 'gzip, deflate'
-    if 'user-agent' in headers:
-      headers['user-agent'] += ' '
+    if 'x-goog-api-client' in headers:
+      headers['x-goog-api-client'] += ' '
     else:
-      headers['user-agent'] = ''
-    headers['user-agent'] += 'google-api-python-client/%s (gzip)' % __version__
```

Comment:
Do we know why (gzip) was here?  Was that a bug?

Reply:
It looks like it was added here https://codereview.appspot.com/8288046/patch/1/3 in response to https://stackoverflow.com/questions/7140035/http-server-not-honoring-accept-encoding-gzip-unless-user-agent-is-known-browse

Reply:
I think we may be able to remove it eventually. Some of the discovery clients don't seem to use it. See https://github.com/googleapis/google-api-go-client/blob/8d96cf9a63b14ab803f9f14ed430221d629476ce/googleapi/googleapi.go#L53

I'll re-add it for now since the primary focus of this PR is adding in the x-goog-api-client header.

Reply:
Ok.  I think you are right that it is not needed, but makes sense to look at changing that separately.